src/libumf.map

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+UMF_1.0 {
+    global:
+        umfFree;
+        umfGetLastFailedMemoryProvider;
+        umfMemoryProviderAlloc;
+        umfMemoryProviderAllocationMerge;
+        umfMemoryProviderAllocationSplit;
+        umfMemoryProviderCreate;
+        umfMemoryProviderDestroy;
+        umfMemoryProviderFree;
+        umfMemoryProviderGetLastNativeError;
+        umfMemoryProviderGetMinPageSize;
+        umfMemoryProviderGetName;
+        umfMemoryProviderGetRecommendedPageSize;
+        umfMemoryProviderPurgeForce;
+        umfMemoryProviderPurgeLazy;
+        umfPoolAlignedMalloc;
+        umfPoolByPtr;
+        umfPoolCalloc;
+        umfPoolCreate;
+        umfPoolDestroy;
+        umfPoolFree;
+        umfPoolGetLastAllocationError;
+        umfPoolGetMemoryProvider;
+        umfPoolMalloc;
+        umfPoolMallocUsableSize;
+        umfPoolRealloc;
+        umfOsMemoryProviderOps;
```

Comment:
what if we disable OS provider in cmake?

Reply:
on linux, that's fine, if there's no such symbol in the .so linker will just ignore it

Reply:
ok but do we plan to generate these files in runtime in the future?

Reply:
Iit depends on whether we will need to support enabling/disabling different features. For example, OS provider (but I think, eventually, we'll want OS provider to always be enabled.